Sources/ArcGISToolkit/Components/OfflineMapAreasView/PreplannedListItemView.swift

```diff
@@ -0,0 +1,101 @@
+                if let thumbnail = preplannedMapModel.preplannedMapArea.portalItem.thumbnail {
+                    LoadableImageView(loadableImage: thumbnail)
+                        .frame(width: 64, height: 44)
+                }
+                VStack(alignment: .leading, spacing: 2) {
+                    Text(preplannedMapArea.portalItem.title)
+                        .font(.headline)
+                    if !preplannedMapArea.portalItem.description.isEmpty {
+                        Text(preplannedMapArea.portalItem.description)
+                            .font(.caption)
+                            .foregroundColor(.secondary)
+                            .lineLimit(2)
+                    }
+                    if let error {
+                        Text(error.localizedDescription)
+                            .foregroundColor(.red)
+                    }
+                }
+                Spacer()
+                switch preplannedMapModel.result {
+                case .success:
+                    Image(systemName: "checkmark.circle.fill")
+                case .failure(let error):
+                    Image(systemName: "exclamationmark.circle")
+                        .foregroundColor(.red)
+                        .task {
+                            self.error = error
+                        }
+                case .none:
+                    if let canDownload {
+                        if !canDownload {
+                            // Map is still packaging.
+                            VStack {
+                                Image(systemName: "clock.badge.xmark")
+                                Text("packaging")
+                                    .font(.footnote)
+                            }
+                            .foregroundColor(.secondary)
+                        } else {
+                            // Map package is available for download.
+                            Image(systemName: "arrow.down.circle")
+                                .foregroundColor(.accentColor)
+                        }
+                    } else {
+                        ProgressView()
+                            .frame(maxWidth: .infinity)
+                    }
+                }
+            }
+            .onReceive(preplannedMapModel.preplannedMapArea.$loadStatus) { status in
+                let packagingStatus = preplannedMapModel.preplannedMapArea.packagingStatus
+                if status == .failed || packagingStatus == .processing {
+                    // If the preplanned map area fails to load, it may not be packaged.
+                    canDownload = false
+                } else if packagingStatus == .complete || packagingStatus == nil {
+                    // Otherwise, check the packaging status to determine if the map area is
+                    // available to download.
+                    canDownload = true
+                }
+            }
```

Comment:
I cleaned up the logic. I left the conditional `nil` check for now because there are cases where both `status = .loaded` and `packagingStatus = nil` are true.

Reply:
Since the `nil` status check is necessary for legacy webmaps that have incomplete metadata, I will leave the `nil` check. I added a documentation comment as well.